net/redisclient_test.go

```diff
@@ -156,7 +147,9 @@ func TestRedisClient(t *testing.T) {
 				}
 			}
 
-			go func() { <-ch }() // create client will block
+			if ch != nil {
+				go func() { <-ch }() // create client will block
+			}
```

Comment:
I think we could create a buffered channel `ch = make(chan struct{}, 1)` to make it cleaner and unblock first `AddrUpdater` call.

Reply:
I tested the suggestion and it does not work

Reply:
Yeah, would need to consume first update produced by NewRedisRingClient:
```diff
diff --git a/net/redisclient_test.go b/net/redisclient_test.go
index e648a40f..72a8dc7a 100644
--- a/net/redisclient_test.go
+++ b/net/redisclient_test.go
@@ -149,14 +149,13 @@ func TestRedisClient(t *testing.T) {
                        if tt.options.AddrUpdater != nil {
                                orig = tt.options.AddrUpdater()
                                // test background update
-                               ch = make(chan struct{})
+                               ch = make(chan struct{}, 1)
                                tt.options.AddrUpdater = func() []string {
                                        ch <- struct{}{}
                                        return []string{redisAddr, redisAddr2}
                                }
                        }
 
-                       go func() { <-ch }() // create client will block
                        cli := NewRedisRingClient(tt.options)
                        defer func() {
                                if !cli.closed {
@@ -170,6 +169,8 @@ func TestRedisClient(t *testing.T) {
                        }
 
                        if tt.options.AddrUpdater != nil {
+                               <-ch // read update produced by NewRedisRingClient
+
                                if !cmp.Equal(orig, []string{redisAddr}) {
                                        t.Errorf("Failed to get addr: %v", cmp.Diff(orig, []string{redisAddr}))
                                }
```

Lets keep it like it is.